examples/qualcomm/llama2/llama_qaihub.py

```diff
@@ -0,0 +1,232 @@
+        required=True,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--seq_len",
+        help="ouput sequence length for llama2",
+        default=128,
+        type=int,
+    )
+
+    parser.add_argument(
+        "--temperature",
+        help="sampling temperature for llama2",
+        default=0.8,
+        type=float,
+    )
+
+    parser.add_argument(
+        "--prompt",
+        help="user prompts for llama2",
+        required=True,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--pre_gen_pte",
+        help="folder path to pre-compiled ptes",
+        default=None,
+        type=str,
+    )
+
+    args = parser.parse_args()
+    target_names = (
+        [
+            f"llama_v2_7b_chat_quantized_Llama2_PromptProcessor_{i}_Quantized.bin"
+            for i in range(1, 5)
+        ]
+        if args.use_prompt_processor
+        else [
+            f"llama_v2_7b_chat_quantized_Llama2_TokenGenerator_{i}_Quantized.bin"
+            for i in range(1, 5)
+        ]
+    )
+
+    # common part for compile & inference
+    backend_options = generate_htp_compiler_spec(
+        use_fp16=False,
+        use_multi_contexts=True,
+    )
+    compiler_specs = generate_qnn_executorch_compiler_spec(
+        soc_model=eval(f"QcomChipset.{args.model}"),
+        backend_options=backend_options,
+        is_from_context_binary=True,
+    )
+
+    if args.pre_gen_pte is None:
+        # create custom operators as context loader
+        bundle_programs = []
+        for i, target in enumerate(target_names):
+            file_name = f"{args.context_binaries}/{target}"
+            bundle_programs.append(from_context_binary(file_name, f"ctx_loader_{i}"))
+        # lower with QnnBackend
+        lowered_modules = []
+        for prog in bundle_programs:
+            lowered_modules.append(
+                to_backend("QnnBackend", prog["edge_program"], compiler_specs)
+            )
+        # setup spill-fill buffer for relieving runtime memory usage
+        canonicalize_program(lowered_modules)
+        # export pte files
+        pte_name, pte_files = "qaihub_llama7b", []
+        for i, _ in enumerate(target_names):
+            memory_planning_pass = MemoryPlanningPass(
+                memory_planning_algo="greedy",
+                alloc_graph_input=False,
+                alloc_graph_output=False,
+            )
+            pte_files.append(f"{args.artifact}/{pte_name}_{i}.pte")
+            with open(pte_files[-1], "wb") as file:
+                file.write(
+                    lowered_modules[0].buffer(memory_planning=memory_planning_pass)
+                )
+            # gc for reducing host memory consuming
+            bundle_programs.pop(0)
+            lowered_modules.pop(0)
+            gc.collect()
+    else:
+        pte_name = "qaihub_llama7b"
+        pte_files = [f"{args.pre_gen_pte}/{pte_name}_{i}.pte" for i in range(4)]
+
+    if args.compile_only:
+        exit(0)
```

Comment:
nit: create a `main()` function, replace this with `return`

Reply:
Done, thanks.